packages/react/src/components/CommandCenter/create.tsx

```diff
@@ -0,0 +1,146 @@
+import { observer } from "mobx-react-lite";
+import { CommandCenter as CommandCenterComponent } from "./CommandCenter";
+import { SuggestionsStore } from "./SuggestionsStore";
+import { CommandCenterSuggestions as CommandCenterSuggestionsComponent } from "./Suggestions";
+import { useEffect, useRef } from "react";
+import { Input } from "./Input";
+import { useKeybinding } from "@/hooks/keyboardEventHandler";
+import { KEY_BINDINGS } from "@/constants/keybindings";
+import { Suggestion } from "@/constants/commandTypes";
+import { ProjectContextType, useProjectContext } from "@/contexts/Project/useProjectContext";
+import { InputStore } from "./InputStore";
+import { CommandCenterStore } from "./CommandCenterStore";
+
+
+const createCommandCenterInput = ({ suggestionsStore, inputStore, onSuggestionSelect }: {
```

Comment:
Each create function can have vi-tests written for it which would essentially be the equivalent of integration tests. We'd be testing the integration of different parts that come together within a create function. Each create function accepts dependencies which make it easier to test in isolation.